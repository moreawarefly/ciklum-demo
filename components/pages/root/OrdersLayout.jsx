import React, { useState } from 'react';
import styled, { css } from 'styled-components';
import AddOrderForm from './AddOrderForm';
import OrdersTable from './OrdersTable';
import { addOrder } from '../../../services/ordersStorageApi';

const mediaDesktop = (...args) => css`
  @media (min-width: 768px) {
    ${css(...args)}
  }
`;

const Main = styled.div`
  ${mediaDesktop`
    display: flex;
    justify-content: center;
  `}
`;

const FormWrapper = styled.div`
  ${mediaDesktop`
    width: 280px;
    padding-right: 30px;
    margin-right: 30px;
    border-right: 1px solid #ccc;
  `}
`;

const OrdersWrapper = styled.div`
  padding-top: 1rem;
  ${mediaDesktop`
    padding-top: 0;
    flex-grow: 1;
    max-width: 600px;
  `}
`;

function OrdersLayout() {
  const [newOrder, setNewOrder] = useState();

  function addOrderHandler(order) {
    addOrder(order);
    setNewOrder(order);
  }

  return (
    <Main>
      <FormWrapper>
        <AddOrderForm
          addOrderHandler={addOrderHandler}
        />
      </FormWrapper>
      <OrdersWrapper>
        <OrdersTable
          newOrder={newOrder}
        />
      </OrdersWrapper>
    </Main>
  );
}

export default OrdersLayout;
