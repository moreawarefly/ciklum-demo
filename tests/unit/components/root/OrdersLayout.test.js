import React from 'react';
import renderer from 'react-test-renderer';
import OrdersLayout from '../../../../components/pages/root/OrdersLayout';

describe('OrdersLayout', () => {
  beforeEach(() => {
    const fakeData = JSON.stringify([{"pair":"btcusd","price_precision":5,"initial_margin":"30.0","minimum_margin":"15.0","maximum_order_size":"2000.0","minimum_order_size":"0.004","expiration":"NA","margin":true}]);
    fetch.resetMocks();
    fetch.mockResponse(fakeData);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<OrdersLayout />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
