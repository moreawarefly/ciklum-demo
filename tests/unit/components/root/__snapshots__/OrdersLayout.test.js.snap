// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersLayout renders correctly 1`] = `
<div
  className="OrdersLayoutstyles__Main-sc-3st4fd-0 dzBGOA"
>
  <div
    className="OrdersLayoutstyles__FormWrapper-sc-3st4fd-1 csUZIe"
  >
    <div>
      Loading
    </div>
  </div>
  <div
    className="OrdersLayoutstyles__OrdersWrapper-sc-3st4fd-2 cixaoq"
  >
    <div
      className="OrdersTablestyles__NoRecordsContainer-sc-1758ceg-5 jJqdFU"
    >
      No orders to be displayed.
    </div>
  </div>
</div>
`;
